### PR TITLE
Add key event property

### LINKED ISSUE #180: Add the “key” property to events
Syn currently emits events with the [which](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which) property, which has been deprecated in favor of the [key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) property.

Syn sets `which` here: https://github.com/bitovi/syn/blob/37814415f59a3e8b75770d5dd010263d8d63963a/src/key.js#L404-L409

### PR DESCRIPTION
**Pull request by cherifGsoul**
This adds `key` property to the event emitted by `syn`.

It adds the use of `KeyboardEvent` constructor in order to pass the `key` property to the event instance, `key` property is read-only.

Edit: I updated the code by adding a mapping from the available [keycodes](https://github.com/bitovi/syn/blob/master/src/key.js#L181) to the correct `KeyboardEvent` key value.

Closes #180

### CODE REVIEW

File: src/key.js [hunk 1]
```diff
@@ -404,8 +404,10 @@ h.extend(syn.key, {
 		// all current browsers have which property to normalize keyCode/charCode
 		if (result.keyCode) {
 			result.which = result.keyCode;
+			result.key = key;
 		} else {
 			result.which = result.charCode;
+			result.key = key;
```
Comment (chasenlehara) [inline src/key.js]: This can be done once instead of repeating the logic across both conditional branches.

File: test/key_test.js [hunk 2]
```diff
@@ -721,3 +721,43 @@ QUnit.test("typing in an input type=number works", function() {
...
+	});
+});
+
+QUnit.test("ctrl key", function () {
+	stop();
+
+	var keyIsDown = false;
+	st.binder("key", "keydown", function (ev) {
+		keyIsDown = ev.ctrlKey;
+		ok(ev.key === 'ctrl', "key is normalized");
```
Comment (chasenlehara) [inline test/key_test.js]: This doesn’t look right to me. According to [these docs](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values), I think it should be `"Control"`.

Comment (cherifGsoul) [reply test/key_test.js]: Thank you for the catch @chasenlehara, I missed this docs.
It is `"Control"` for sure.
Screen Shot 2020-03-26 at 5 29 54 PM

Comment (cherifGsoul) [reply test/key_test.js]: I updated the code @chasenlehara !

**Pull request merged**